portal-bridge/src/api/execution.rs

```diff
@@ -57,21 +66,28 @@ impl ExecutionApi {
                 };
                 let base_el_endpoint =
                     Url::parse(&endpoint).expect("to be able to parse static base el endpoint url");
-                Config::new()
+                let client: Client = Config::new()
                     .add_header("Content-Type", "application/json")?
                     .add_header("CF-Access-Client-Id", PANDAOPS_CLIENT_ID.to_string())?
                     .add_header(
                         "CF-Access-Client-Secret",
                         PANDAOPS_CLIENT_SECRET.to_string(),
                     )?
                     .set_base_url(base_el_endpoint)
-                    .try_into()?
+                    .try_into()?;
+                client.with(rate_limit).with(Retry::default())
+            }
+            Provider::Url(url) => {
+                let client: Client = Config::new()
+                    .add_header("Content-Type", "application/json")?
+                    .set_base_url(url.clone())
+                    .try_into()?;
+                client.with(rate_limit).with(Retry::default())
+            }
+            Provider::Test => {
+                let client: Client = Config::new().try_into()?;
+                client.with(rate_limit).with(Retry::default())
```

Comment:
I'm not sure we need a rate limit for reading data from a test file.... but if so, you can move the..
```
client.with(rate_limit).with(Retry::default())
```
line below, since it applies in each case, and then you don't have to repeat it 3 times inside each clause